framework/moose.mk

```diff
@@ -312,11 +312,12 @@ $(foreach srcsubdir,$(unity_srcsubdirs),$(eval $(call unity_file_rule,$(call uni
 
 app_unity_srcfiles := $(foreach srcsubdir,$(unity_srcsubdirs),$(call unity_unique_name,$(unity_src_dir),$(FRAMEWORK_DIR),$(srcsubdir)))
 
-#$(info $(app_unity_srcfiles))
-
 unity_srcfiles += $(app_unity_srcfiles)
 
-moose_srcfiles    := $(app_unity_srcfiles) $(shell find $(non_unity_srcsubdirs) -maxdepth 1 -regex "[^\#~]*\.C") $(shell find $(filter-out %/src,$(moose_SRC_DIRS)) -regex "[^\#~]*\.C")
+ifneq ($(non_unity_srcsubdirs),)
+	app_nonunity_srcfiles = $(shell find $(non_unity_srcsubdirs) -maxdepth 1 -regex "[^\#~]*\.C") $(shell find $(filter-out %/src,$(moose_SRC_DIRS)) -regex "[^\#~]*\.C")
+endif
```

Comment:
what does find do when $(non_unity_srcsubdirs) is empty? seems it is finding some contrib here

Reply:
i think we could have an "else" branch with no maxdepth used in the find and avoid the error

Reply:
When `non_unity_srcsubdirs` is empty, `find` emits the error message (which we are trying to fix here).

Reply:
and it does not find anything? it just errors?

Reply:
i guess so from local testing. umm so why are contribs not building

Reply:
I can't seem to reproduce the error locally.

Reply:
I mean some build boxes work, while others don't...

Reply:
trying something

Reply:
I assume you finished "trying something"... I'll look more into this now.

Reply:
I think It works now. do you approve my 1 line change?

Reply:
sure -- if it works it works...